services/document-manager/backend/app/services/object_store_storage_service.py

```diff
@@ -249,4 +253,86 @@ def upload_zip_file(self, file_data, key, file_size, progress_callback):
             
             return True
         except ClientError as e:
-            raise Exception(f'Failed to upload the file: {e}')
\ No newline at end of file
+            raise Exception(f'Failed to upload the file: {e}')
+
+    def __download_in_chunks(self, key):
+        # Generator function to download file in chunks of 8MB size
+
+        # Get the object from S3
+        response = self._client.get_object(Bucket=Config.OBJECT_STORE_BUCKET, Key=key)
+
+        # Read the object in chunks and yield them one by one
+        for chunk in response['Body'].iter_chunks(chunk_size=8 * 1024 * 1024):
+            yield chunk
+
+    def create_zip_and_upload(self, key, document_paths, pretty_paths, progress_callback):
+        # Calculate the total size of all the files
+        total_size = sum(
+            self._client.head_object(Bucket=Config.OBJECT_STORE_BUCKET, Key=path)['ContentLength'] for path in
+            document_paths)
+
+        def local_files():
+            # Get the current timestamp
+            now = datetime.now()
+
+            def contents(path):
+                # Generator to read the file in chunks from S3
```

Comment:
Even having writting a lot of this, I find it to be difficult code to read through.  Have a couple thoughts on making the code comments a bit clearer for anyone who might deal with this down the road.

```suggestion
                # A generator function to slowly take data (in chunks) from S3 without loading the whole file in the memory.
```